Azure/sonic-mgmt
# [arp]: Add ARP pytests to support dual ToR use cases

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>
### Description of PR


Summary:
* Adds ARP test cases needed for dual ToR support (IPv4 GARP, IPv4 and IPv6 proxy ARP)
* Refactor several common utility fixtures/methods

### Type of change




### Approach
#### What is the motivation for this PR?
Dual ToR setups require support for learning gratuitous ARP (GARP) messages, as well as sending proxy ARP/NDP messages for IPs within the VLAN subnet

#### How did you do it?
GARP:
1. Enable GARP on the DUT
2. Send an ARP reply from the PTF to the DUT
3. Check the neighbor table on the DUT

Proxy ARP:
1. Enable proxy ARP on the DUT
2. Assign an IP address from the DUT VLAN's subnet to an interface on the PTF which is connected to that VLAN
3. Attempt to ping the a different IP address still in the VLAN's subnet from the PTF via the above interface
4. Check the PTF's neighbor table

#### How did you verify/test it?
* Run new test cases and check that they pass
* Run existing ARP test cases and verify that they pass

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?
T0 and dual ToR topologies

### Documentation 

IPv4/v6 tests in progress
